test/unit/core/selectors.ts

```diff
@@ -1,8 +1,34 @@
+import { MAX_RECORDS } from '../../../src/core/adapters/search';
 import Selectors from '../../../src/core/selectors';
 import suite from '../_suite';
 
 suite('selectors', ({ expect, stub }) => {
 
+  describe('searchRequest()', () => {
+    it('should decrease page size to prevent exceeding MAX_RECORDS', () => {
+      stub(Selectors, 'fields');
+      stub(Selectors, 'query');
+      stub(Selectors, 'collection');
+      stub(Selectors, 'selectedRefinements');
+      stub(Selectors, 'sort');
+      stub(Selectors, 'requestSort');
+
+      const originalPageSize = MAX_RECORDS - 1;
+      const state: any = {
```

Comment:
I think you should be able to avoid passing a "real" state to this test if you stub `Selectors.pageSize()`,
good otherwise